backend/tracim_backend/lib/event.py

```diff
@@ -0,0 +1,287 @@
+        LiveMessageBuilder.session = db_session
+        LiveMessageBuilder.config = self._config
+        queue.enqueue(
+            LiveMessageBuilder.publish_messages_for_event, self._event_schema.dump(event).data,
+        )
+
+
+class LiveMessageBuilder:
+
+    _event_schema = EventSchema()
+    session = None
+    config = None
+
+    @classmethod
+    def _session(cls) -> TracimSession:
+        # TODO S.G 2020-05-07: acquire this from the RQ worker's context
+        assert cls.session
+        return cls.session
+
+    @classmethod
+    def _config(cls) -> CFG:
+        # TODO SG 2020-05-07: acquire this from the RQ worker's context
+        assert cls.config
+        return cls.config
+
+    @classmethod
+    def publish_messages_for_event(cls, event_dict: typing.Dict[str, typing.Any]) -> None:
+        load_result = cls._event_schema.load(event_dict)
+        if load_result.errors:
+            logger.error(
+                cls, "Error while loading event from dict: {}".format(load_result.errors),
+            )
+            return
+        event = Event(**load_result.data)
+        if event.entity_type == EntityType.USER:
+            receiver_ids = cls._get_user_event_receiver_ids(event)
+        elif event.entity_type == EntityType.WORKSPACE:
+            receiver_ids = cls._get_workspace_event_receiver_ids(event)
+        elif event.entity_type == EntityType.CONTENT:
+            receiver_ids = cls._get_workspace_event_receiver_ids(event)
+        elif event.entity_type == EntityType.WORKSPACE_USER_ROLE:
+            receiver_ids = cls._get_user_event_receiver_ids(event)
+
+        # The author is not notified of the change (?)
+        try:
+            receiver_ids -= set([event.author["user_id"]])
+        except (AttributeError, KeyError):
+            # No author
+            pass
+
+        messages = [
+            Message(receiver_id=receiver_id, event_id=event.event_id, sent=datetime.utcnow())
+            for receiver_id in receiver_ids
+        ]
+        cls._session().add_all(messages)
+
+    @classmethod
+    def _get_user_event_receiver_ids(cls, event: Event) -> typing.Set[int]:
+        user_api = UserApi(current_user=None, session=cls._session(), config=cls._config())
+        receiver_ids = set(user_api.get_user_ids_from_profile(Profile.ADMIN))
+        if event.user:
+            receiver_ids.add(event.user["user_id"])
+        return receiver_ids
+
+    @classmethod
+    def _get_workspace_event_receiver_ids(cls, event: Event) -> typing.Set[int]:
+        user_api = UserApi(current_user=None, session=cls._session(), config=cls._config())
+        administrators = user_api.get_user_ids_from_profile(Profile.ADMIN)
+        role_api = RoleApi(current_user=None, session=cls._session(), config=cls._config())
+        workspace_members = role_api.get_workspace_member_ids(event.workspace["workspace_id"])
+        return set(administrators).union(set(workspace_members))
```

Comment:
```suggestion
        return set(administrators).union(workspace_members)
```

```suggestion
        return set(administrators + workspace_members)
```
(works with Python lists, not sure about the types of `administrators` and `workspace_members`)

Reply:
I didn't know that `union` worked with a list as parameter, thank you for the tip.